### PR TITLE
Some analysis fixes

### PR DESCRIPTION
**Pull request by craig65535**
I ran a Coverity analysis on ZipArchive, and it revealed a number of issues. This fixes ~4~ 19 of them.

### CODE REVIEW

File: SSZipArchive/minizip/unzip.c [hunk 1]
```diff
@@ -340,7 +340,7 @@ static int unzSearchCentralDir64(const zlib_filefunc64_32_def *pzlib_filefunc_de
 
 static unzFile unzOpenInternal(const void *path, zlib_filefunc64_32_def *pzlib_filefunc64_32_def)
 {
-    unz64_internal us;
+    unz64_internal us = { 0 };
```
Comment (craig65535) [inline SSZipArchive/minizip/unzip.c]: The read happens on line 494:
```
*s = us;
```

File: SSZipArchive/minizip/zip.c [hunk 2]
```diff
@@ -891,7 +891,7 @@ extern zipFile ZEXPORT zipOpen3_64(const void *path, int append, uint64_t disk_s
 {
     if (pzlib_filefunc_def != NULL)
     {
-        zlib_filefunc64_32_def zlib_filefunc64_32_def_fill;
+        zlib_filefunc64_32_def zlib_filefunc64_32_def_fill = { 0 };
```
Comment (craig65535) [inline SSZipArchive/minizip/zip.c]: Write is on line 628, in zipOpen4:
```
ziinit.z_filefunc = *pzlib_filefunc64_32_def;
```

File: SSZipArchive/minizip/zip.c [hunk 3]
```diff
@@ -865,7 +865,7 @@ extern zipFile ZEXPORT zipOpen2_64(const void *path, int append, const char **gl
...
```
Comment (craig65535) [inline SSZipArchive/minizip/zip.c]: Write is on line 628, in zipOpen4:
```
ziinit.z_filefunc = *pzlib_filefunc64_32_def;
```

File: SSZipArchive/minizip/unzip.c [hunk 4]
```diff
@@ -516,7 +516,7 @@ extern unzFile ZEXPORT unzOpen2_64(const void *path, zlib_filefunc64_def *pzlib_
 {
     if (pzlib_filefunc_def != NULL)
     {
-        zlib_filefunc64_32_def zlib_filefunc64_32_def_fill;
+        zlib_filefunc64_32_def zlib_filefunc64_32_def_fill = { 0 };
```
Comment (craig65535) [inline SSZipArchive/minizip/unzip.c]: Read is on line 363, in unzOpenInternal:
```
us.z_filefunc = *pzlib_filefunc64_32_def;
```

**Comment by craig65535** [comment]
@Coeur addressed 3 more issues

File: SSZipArchive/minizip/unzip.c [hunk 5]
```diff
@@ -362,6 +362,13 @@ static unzFile unzOpenInternal(const void *path, zlib_filefunc64_32_def *pzlib_f
     else
         us.z_filefunc = *pzlib_filefunc64_32_def;
 
+    if (!(us.z_filefunc.zopen32_file || us.z_filefunc.zfile_func64.zopen64_file) ||
```
Comment (craig65535) [inline SSZipArchive/minizip/unzip.c]: This fix is perhaps not necessary, but it's good practice and silences 5 "FORWARD_NULL" issues. It ensures that at least one of each pair of function pointers is not NULL.

File: SSZipArchive/minizip/zip.c [hunk 6]
```diff
@@ -627,6 +627,13 @@ extern zipFile ZEXPORT zipOpen4(const void *path, int append, uint64_t disk_size
     else
         ziinit.z_filefunc = *pzlib_filefunc64_32_def;
 
+    if (!(ziinit.z_filefunc.zopen32_file || ziinit.z_filefunc.zfile_func64.zopen64_file) ||
```
Comment (craig65535) [inline SSZipArchive/minizip/zip.c]: This fix is perhaps not necessary, but it's good practice and silences 7 "FORWARD_NULL" issues. It ensures that at least one of each pair of function pointers is not NULL.

**Comment by craig65535** [comment]
@Coeur Are any changes needed here?

**Comment by Coeur** [comment]
I was in Japan lately. I'm back home now and I'll look at it closely this weekend. Note that you may want to replicate your improvements for https://github.com/coeur/minizip/

But I need to make sure `unzClose(zip)` doesn't break anything, as recently I recall someone made a similar change on Minizip2 (https://github.com/nmoinvaz/minizip/pull/206) and a day later he noticed he broke something (https://github.com/nmoinvaz/minizip/pull/207).

**Comment by craig65535** [comment]
@Coeur I opened a pull request against coeur/minizip: https://github.com/Coeur/minizip/pull/6

File: SSZipArchive/SSZipArchive.m [hunk 7]
```diff
@@ -50,27 +50,30 @@ + (BOOL)isFilePasswordProtectedAtPath:(NSString *)path {
...
             }
             
             unzCloseCurrentFile(zip);
             ret = unzGoToNextFile(zip);
         } while (ret == UNZ_OK);
     }
-    
-    return NO;
+
+    unzClose(zip);
```
Comment (craig65535) [inline SSZipArchive/SSZipArchive.m]: @Coeur I think unzClose is OK here. This function is "self-contained" - it opens a file and determines if it is password-protected.

File: SSZipArchive/SSZipArchive.m [hunk 8]
```diff
@@ -128,18 +135,20 @@ + (BOOL)isPasswordValidForArchiveAtPath:(NSString *)path password:(NSString *)pw
...
             }
             
             unzCloseCurrentFile(zip);
             ret = unzGoToNextFile(zip);
         } while (ret == UNZ_OK);
     }
     
-    // No password required
-    return YES;
+    unzClose(zip);
```
Comment (craig65535) [inline SSZipArchive/SSZipArchive.m]: @Coeur This function is also "self-contained"

**Comment by Coeur** [comment]
I've merged the minizip changes. I'll review and merge the rest later, no worry.

**Comment by craig65535** [comment]
@Coeur updated to match your https://github.com/Coeur/minizip/pull/6 changes

**Comment by craig65535** [comment]
@Coeur This should be good now that it matches Coeur/minizip#6 ? Let me know if any other changes are required.

**Comment by jhudsonWA** [comment]
I'll take a look at this after the holidays. Just want to let you know you were not forgotten. Thanks for the PR.

**Comment by craig65535** [comment]
@Coeur @jhudsonWA Just checking in to see what your status is on this PR

**Comment by jhudsonWA** [comment]
Let me pull in the latest minizip first, then we can merge in the other changes. They are on version 2, so this could be an easy merge or not, so a bit more patience.

**Comment by Coeur** [comment]
Sorry for the wait.
I've squashed your changed to solve the conflict and prevent the history changes on minizip.

**Pull request merged**